### ISSUE TITLE
TF2.0beta1 - Not JSON Serializable when using tf.keras.experimental.export_saved_model

### ISSUE THREAD

**Issue by yl1991** [tensorflow/tensorflow#31498]
Most code is from one of the tensorflow 2.0 beta guides: [Writing layers and models with TensorFlow Keras](https://www.tensorflow.org/beta/guide/keras/custom_layers_and_models#putting_it_all_together_an_end-to-end_example)

**Describe the current behavior**

TypeError: ('Not JSON Serializable:', b'\n\x06Square\x12\x06Square\x1a\x0fz_mean/Identity*\x07\n\x01T\x12\x020\x01')

**Describe the expected behavior**

Save model correctly.

**Code to reproduce the issue**
```
import tensorflow as tf
from tensorflow.keras import layers

# Get training data.
(x_train, _), _ = tf.keras.datasets.mnist.load_data()
x_train = x_train.reshape(60000, 784).astype('float32') / 255

original_dim = 784
intermediate_dim = 64
latent_dim = 32

def sampling(inputs):
    z_mean, z_log_var = inputs
    batch = tf.shape(z_mean)[0]
    dim = tf.shape(z_mean)[1]
    epsilon = tf.keras.backend.random_normal(shape=(batch, dim))
    return z_mean + tf.exp(0.5 * z_log_var) * epsilon    

# Define encoder model.
original_inputs = tf.keras.Input(shape=(original_dim,), name='encoder_input')
x = layers.Dense(intermediate_dim, activation='relu')(original_inputs)
z_mean = layers.Dense(latent_dim, name='z_mean')(x)
z_log_var = layers.Dense(latent_dim, name='z_log_var')(x)
z = tf.keras.layers.Lambda(sampling)((z_mean, z_log_var))
encoder = tf.keras.Model(inputs=original_inputs, outputs=z, name='encoder')

# Define decoder model.
latent_inputs = tf.keras.Input(shape=(latent_dim,), name='z_sampling')
x = layers.Dense(intermediate_dim, activation='relu')(latent_inputs)
outputs = layers.Dense(original_dim, activation='sigmoid')(x)
decoder = tf.keras.Model(inputs=latent_inputs, outputs=outputs, name='decoder')

# Define VAE model.
outputs = decoder(z)
vae = tf.keras.Model(inputs=original_inputs, outputs=outputs, name='vae')

# Add KL divergence regularization loss.
kl_loss = - 0.5 * tf.reduce_mean(
    z_log_var - tf.square(z_mean) - tf.exp(z_log_var) + 1)
vae.add_loss(kl_loss)

# Train.
optimizer = tf.keras.optimizers.Adam(learning_rate=1e-3)
vae.compile(optimizer, loss=tf.keras.losses.MeanSquaredError())
vae.fit(x_train, x_train, epochs=3, batch_size=64)

# Save model.
tf.keras.experimental.export_saved_model(vae, 'vae_functional_saved_model')
```
**Other info / logs**
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-4eb0216166eb> in <module>
----> 1 tf.keras.experimental.export_saved_model(vae, 'vae_functional_saved_model')

~/.miniconda/envs/tf2/lib/python3.7/site-packages/tensorflow/python/keras/saving/saved_model.py in export_saved_model(model, saved_model_path, custom_objects, as_text, input_signature, serving_only)
    167 
    168   try:
--> 169     _export_model_json(model, saved_model_path)
    170   except NotImplementedError:
    171     logging.warning('Skipped saving model JSON, subclassed model does not have '

~/.miniconda/envs/tf2/lib/python3.7/site-packages/tensorflow/python/keras/saving/saved_model.py in _export_model_json(model, saved_model_path)
    175 def _export_model_json(model, saved_model_path):
    176   """Saves model configuration as a json string under assets folder."""
--> 177   model_json = model.to_json()
    178   model_json_filepath = os.path.join(
    179       saved_model_utils.get_or_create_assets_dir(saved_model_path),

~/.miniconda/envs/tf2/lib/python3.7/site-packages/tensorflow/python/keras/engine/network.py in to_json(self, **kwargs)
   1447     model_config = self._updated_config()
   1448     return json.dumps(
-> 1449         model_config, default=serialization.get_json_type, **kwargs)
   1450 
   1451   def to_yaml(self, **kwargs):

~/.miniconda/envs/tf2/lib/python3.7/json/__init__.py in dumps(obj, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, default, sort_keys, **kw)
    236         check_circular=check_circular, allow_nan=allow_nan, indent=indent,
    237         separators=separators, default=default, sort_keys=sort_keys,
--> 238         **kw).encode(obj)
    239 
    240 

~/.miniconda/envs/tf2/lib/python3.7/json/encoder.py in encode(self, o)
    197         # exceptions aren't as detailed.  The list call should be roughly
    198         # equivalent to the PySequence_Fast that ''.join() would do.
--> 199         chunks = self.iterencode(o, _one_shot=True)
    200         if not isinstance(chunks, (list, tuple)):
    201             chunks = list(chunks)

~/.miniconda/envs/tf2/lib/python3.7/json/encoder.py in iterencode(self, o, _one_shot)
    255                 self.key_separator, self.item_separator, self.sort_keys,
    256                 self.skipkeys, _one_shot)
--> 257         return _iterencode(o, 0)
    258 
    259 def _make_iterencode(markers, _default, _encoder, _indent, _floatstr,

~/.miniconda/envs/tf2/lib/python3.7/site-packages/tensorflow/python/util/serialization.py in get_json_type(obj)
     67     return dict(obj)
     68 
---> 69   raise TypeError('Not JSON Serializable:', obj)

TypeError: ('Not JSON Serializable:', b'\n\x06Square\x12\x06Square\x1a\x0fz_mean/Identity*\x07\n\x01T\x12\x020\x01')
```

**Comment by yl1991** [comment]
The code seems working fine on with tf-nightly-gpu-1.15.0.dev20190809

**Comment by nguerinjr** [comment]
It also seems to work with tf-nightly-2.0-preview-2.0.0.dev20190811. In this version `tf.keras.experimental.export_saved_model` is deprecated. 

``` python
Please use `model.save(..., save_format="tf")` or `tf.keras.models.save_model(..., save_format="tf")`.
```

Although there are some prints that I do not know if they represent an error in saving / loading the optimizer:
``` python
W0811 20:53:33.277865 140585931835200 export_utils.py:182] Export includes no default signature!
W0811 20:53:33.515164 140585931835200 export_utils.py:182] Export includes no default signature!
W0811 20:53:33.587220 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer.beta_1
W0811 20:53:33.587334 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer.beta_2
W0811 20:53:33.587379 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer.decay
W0811 20:53:33.587415 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer.learning_rate
W0811 20:53:33.587450 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-0.kernel
W0811 20:53:33.587482 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-0.bias
W0811 20:53:33.587515 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-1.kernel
W0811 20:53:33.587546 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-1.bias
W0811 20:53:33.587577 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-2.kernel
W0811 20:53:33.587608 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-2.bias
W0811 20:53:33.587643 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-3.layer_with_weights-0.kernel
W0811 20:53:33.587675 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-3.layer_with_weights-0.bias
W0811 20:53:33.587706 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-3.layer_with_weights-1.kernel
W0811 20:53:33.587737 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-3.layer_with_weights-1.bias
W0811 20:53:33.587768 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-0.kernel
W0811 20:53:33.587799 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-0.bias
W0811 20:53:33.587830 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-1.kernel
W0811 20:53:33.587860 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-1.bias
W0811 20:53:33.587891 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-2.kernel
W0811 20:53:33.587921 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-2.bias
W0811 20:53:33.587952 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-3.layer_with_weights-0.kernel
W0811 20:53:33.587983 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-3.layer_with_weights-0.bias
W0811 20:53:33.588013 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-3.layer_with_weights-1.kernel
W0811 20:53:33.588044 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-3.layer_with_weights-1.bias
W0811 20:53:33.588074 140585931835200 util.py:152] A checkpoint was restored (e.g. tf.train.Checkpoint.restore or tf.keras.Model.load_weights) but not all checkpointed values were used. See above for specific issues. Use expect_partial() on the load status object, e.g. tf.train.Checkpoint.restore(...).expect_partial(), to silence these warnings, or use assert_consumed() to make the check explicit. See https://www.tensorflow.org/alpha/guide/checkpoints#loading_mechanics for details.
W0811 20:53:33.801549 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer
W0811 20:53:33.801669 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer.iter
W0811 20:53:33.801716 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer.beta_1
W0811 20:53:33.801752 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer.beta_2
W0811 20:53:33.801786 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer.decay
W0811 20:53:33.801818 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer.learning_rate
W0811 20:53:33.801850 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-0.kernel
W0811 20:53:33.801882 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-0.bias
W0811 20:53:33.801916 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-1.kernel
W0811 20:53:33.801947 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-1.bias
W0811 20:53:33.801977 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-2.kernel
W0811 20:53:33.802012 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-2.bias
W0811 20:53:33.802043 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-3.layer_with_weights-0.kernel
W0811 20:53:33.802074 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-3.layer_with_weights-0.bias
W0811 20:53:33.802104 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-3.layer_with_weights-1.kernel
W0811 20:53:33.802134 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-3.layer_with_weights-1.bias
W0811 20:53:33.802165 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-0.kernel
W0811 20:53:33.802195 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-0.bias
W0811 20:53:33.802225 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-1.kernel
W0811 20:53:33.802255 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-1.bias
W0811 20:53:33.802286 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-2.kernel
W0811 20:53:33.802316 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-2.bias
W0811 20:53:33.802347 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-3.layer_with_weights-0.kernel
W0811 20:53:33.802377 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-3.layer_with_weights-0.bias
W0811 20:53:33.802407 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-3.layer_with_weights-1.kernel
W0811 20:53:33.802437 140585931835200 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'v' for (root).layer_with_weights-3.layer_with_weights-1.bias
W0811 20:53:33.802467 140585931835200 util.py:152] A checkpoint was restored (e.g. tf.train.Checkpoint.restore or tf.keras.Model.load_weights) but not all checkpointed values were used. See above for specific issues. Use expect_partial() on the load status object, e.g. tf.train.Checkpoint.restore(...).expect_partial(), to silence these warnings, or use assert_consumed() to make the check explicit. See https://www.tensorflow.org/alpha/guide/checkpoints#loading_mechanics for details.

Process finished with exit code 0
```

This print also occurs in the loading of the model (which, for compatibility, I've done with the `tf.keras.experimental` version):

``` python
tf.keras.experimental.export_saved_model(vae, 'vae_functional_saved_model')
print(tf.keras.experimental.load_from_saved_model('vae_functional_saved_model'))
```

**Comment by ymodak** [comment]
This is fixed with latest tf-2.0-nightly build  '2.0.0-dev20190812'.
